docs/life-of-a-request.md

```diff
@@ -0,0 +1,200 @@
+# Life of a Request
+
+It can sometimes be confusing to figure out how to write a new feature,
+or debug an existing one. Let's try to follow a request through the
+Zulip codebase, and dive deep into how each part works.
+
+## A request is sent to the server, and handled by Nginx
+
+Everything in Zulip goes through nginx, when deployed in production.
+For the most part you don't need to know how this works, except for when
+it isn't working. Nginx does the first level of routing--deciding which
+application will serve the request (or deciding to serve the request
+itself for static content).
+
+## Nginx secures traffic with SSL
+
+If you visit your zulip server in your browser and discover that your
+traffic isn't being properly encrypted, an nginx configuration is the
+likely culprit.
+
+## Static files are served directly by Nginx
+
+Static files include javascript, css, static assets (like emoji, avatars),
+and user uploads (if stored locally and not on S3).
```

Comment:
I'd probably split this into two sentences just to be clear on what the S3 parenthetical refers to. Something like `Static files include javascript, css, and static assets (like emoji and avatars). They also include user uploads which are stored locally rather than on S3.`